buildkitd/buildkitd.go

```diff
@@ -403,3 +408,21 @@ func isRootlessDocker(ctx context.Context) (bool, error) {
 
 	return strings.Contains(string(output), "rootless"), nil
 }
+
+func supportsPlatform(ctx context.Context) bool {
+	// We can't run scratch, but the error is different depending on whether
+	// --platform is supported or not. This is faster than attempting to run
+	// an actual image which may require downloading.
+	cmd := exec.CommandContext(ctx,
+		"docker", "run", "--rm", platformFlag(), "scratch")
+	output, _ := cmd.CombinedOutput()
+	return bytes.Contains(output, []byte("Unable to find image"))
+}
+
+func platformFlag() string {
+	arch := runtime.GOARCH
+	if runtime.GOARCH == "arm" {
+		arch = "arm/v7"
+	}
+	return fmt.Sprintf("--platform=linux/%s", arch)
```

Comment:
Thank you for not shelling out. Is `arm/v7` a safe assumption?

Reply:
I saw that Docker's libraries do the same - if they see arm, they assume v7. v8 shows up as arm64 so we're safe there.